backend/open_webui/storage/provider.py

```
@@ -115,55 +139,84 @@ def _delete_all_from_s3(self) -> None:
-        return self._get_file_from_local(file_path)
+        """Handles downloading of the file from GCS storage."""
+        try:
+            filename = file_path.removeprefix("gs://").split("/")[1]
+            local_file_path = f"{UPLOAD_DIR}/{filename}"
+            blob = self.bucket.get_blob(filename)
+            blob.download_to_filename(local_file_path)
 
-    def delete_file(self, file_path: str) -> None:
-        """Deletes a file either from S3 or the local file system."""
-        filename = file_path.split("/")[-1]
+            return local_file_path
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)